### ISSUE TITLE
postgres-exporter - pgBackRest query fails in untrusted TLS environment

### ISSUE THREAD

**Issue by lphan-clv** [CrunchyData/postgres-operator#2314]
**Describe the bug**
I have a postgres cluster is working on Azure with metrics enabled. Everything worked fine except for postgres-exporter, it only worked on `replicas deployment` 

**To Reproduce**

1. My own cluster is deployed with this command(*be noticed that all requirements are prepared already: s3, tls, prometheus,..*):
```
pgo create cluster infra \
    --pgbouncer \
    --service-type=LoadBalancer \
    --server-ca-secret=postgresql-ca \
    --server-tls-secret=infra-tls-keypair \
    --pgbouncer-tls-secret=infra-tls-keypair \
    --replication-tls-secret=infra-rep-tls-keypair \
    --pgbouncer-replicas=2 \
    --replica-count=1 \
    --pvc-size=10Gi \
    --pgbackrest-storage-type=s3 \
    --pgbackrest-s3-bucket=<S3-BUCKET> \
    --pgbackrest-s3-endpoint=<S3-ENDPOINT> \
    --pgbackrest-s3-key=<S3-REGION> \
    --pgbackrest-s3-key-secret=<S3-PROJECT> \
    --pgbackrest-s3-region=eu-cenral-1 \
    --pgbackrest-s3-uri-style=path \
    --pgbackrest-s3-verify-tls=false \
    --metrics

```

2. I haved added 2 service monitors from my prometheus operator, they are:

leader-postgres-exporter:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  annotations:
    meta.helm.sh/release-name: prometheus
    meta.helm.sh/release-namespace: cicd
  labels:
    app: prometheus-operator-prometheus
    app.kubernetes.io/managed-by: Helm
    chart: prometheus-operator-8.13.8
    heritage: Helm
    pg-cluster-role: master
    release: prometheus
  name: psql-infra-master
  namespace: cicd
spec:
  endpoints:
  - port: postgres-exporter
  jobLabel: psql-infra-master
  namespaceSelector:
    matchNames:
    - pgo
  selector:
    matchLabels:
      name: infra
      pg-cluster: infra
```

replica-postgres-exporter:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  annotations:
    meta.helm.sh/release-name: prometheus
    meta.helm.sh/release-namespace: cicd
  labels:
    app: prometheus-operator-prometheus
    app.kubernetes.io/managed-by: Helm
    chart: prometheus-operator-8.13.8
    heritage: Helm
    pg-cluster-role: replica
    release: prometheus
  name: psql-infra-replica
  namespace: cicd
spec:
  endpoints:
  - port: postgres-exporter
  jobLabel: psql-infra-replica
  namespaceSelector:
    matchNames:
    - pgo
  selector:
    matchLabels:
      name: infra-replica
      pg-cluster: infra
```

3. From my prometheus, I see leader-postgres-exporter is always unaccessible. Even if I switched role between them, it is  always unaccessible.

![image](https://user-images.githubusercontent.com/67224217/109824913-df062980-7c6b-11eb-83f7-9c576c3e900c.png)

4. Finally I looked at how differences between theirs k8s manifests. I see `database` sidecar from leader pod missed these configurations, `exporter` pods are same:

![image](https://user-images.githubusercontent.com/67224217/109825606-91d68780-7c6c-11eb-8cb4-b40f20f7ce6c.png)
 
I have also tried to run curl inside leader exporter pod, and the result is failed:

![image](https://user-images.githubusercontent.com/67224217/109829435-1a0a5c00-7c70-11eb-8775-428ad2cc6bba.png)

**Expected behavior**
I am not sure it's feature or bug, I just want to have 2 dedicated metrics for both of them.

**Screenshots**
*(Included already)*

**Please tell us about your environment:**

* Operating System: Linux
* Where is this running ( Local, Cloud Provider): Azure
* Storage being used (NFS, Hostpath, Gluster, etc): Azure persistent volumes
* Container Image Tag: registry.developers.crunchydata.com/crunchydata/crunchy-postgres-exporter:centos8-4.6.0
* PostgreSQL Version: registry.developers.crunchydata.com/crunchydata/crunchy-postgres-ha:centos8-13.1-4.6.0
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* Platform Version: 1.17

**Additional context**

**Comment by jkatz** [comment]
I would first upgrade to 4.6.1 as there was a fix for the exporter, though I don't know if that affects you.

However, I ran a nearly identical command to what you provided (I did not use the S3 bits nor the LoadBalancer ServiceType), i.e.:


```
pgo create cluster infra \
    --pgbouncer \
    --service-type=LoadBalancer \
    --server-ca-secret=postgresql-ca \
    --server-tls-secret=hippo.tls \
    --pgbouncer-tls-secret=hippo.tls \
    --replication-tls-secret=hippo.tls \
    --pgbouncer-replicas=2 \
    --replica-count=1 \
    --pvc-size=10Gi \
    --metrics
  ```

and everything came up healthy:

```
kubectl  get pods
NAME                                          READY   STATUS      RESTARTS   AGE
backrest-backup-infra-twbff                   0/1     Completed   0          54s
infra-5c87f85c68-qgxld                        2/2     Running     0          74s
infra-backrest-shared-repo-84bf588b7c-tcj8f   1/1     Running     0          76s
infra-pgbouncer-74b9494875-6bwhd              1/1     Running     0          57s
infra-pgbouncer-74b9494875-984bz              1/1     Running     0          57s
infra-qtxf-5686b758dc-5xkzq                   2/2     Running     0          39s
```

```
$ kubectl logs infra-5c87f85c68-qgxld -c exporter
Wed Mar  3 15:43:30 UTC 2021 INFO: Starting postgres-exporter..
time="2021-03-03T15:43:30Z" level=info msg="Established new database connection to \"127.0.0.1:5432\"." source="postgres_exporter.go:878"
time="2021-03-03T15:43:30Z" level=info msg="Semantic Version Changed on \"127.0.0.1:5432\": 0.0.0 -> 13.2.0" source="postgres_exporter.go:1405"
time="2021-03-03T15:43:31Z" level=info msg="Starting Server: :9187" source="postgres_exporter.go:1672"
```

```

```
$ kubectl logs infra-qtxf-5686b758dc-5xkzq -c exporter
time="2021-03-03T15:44:03Z" level=info msg="Established new database connection to \"127.0.0.1:5432\"." source="postgres_exporter.go:878"
time="2021-03-03T15:44:03Z" level=info msg="Semantic Version Changed on \"127.0.0.1:5432\": 0.0.0 -> 13.2.0" source="postgres_exporter.go:1405"
time="2021-03-03T15:44:03Z" level=info msg="Starting Server: :9187" source="postgres_exporter.go:1672"
```

The diff between the primary and replica deployments have the correct mounts in my environment. It shows that the primary Deployment has both the exporter Secret (which is juts used for bootstrapping) and the init ConfigMap mounted to it. Those are all correct.

I don't see any evidence that there is a bug, though I do suggest upgrading to 4.6.1.

**Comment by lphan-clv** [comment]
@jkatz : thanks for answered that, can you try to get /metrics on both `exporter` containers ? My current version successfully started like your logs, but I can't get metrics on exporter-leader endpoint: "localhost:9187/metrics", while replica is fine. 

```
# leader pod -> doesn't work
kubectl exec -ti infra-5c87f85c68-qgxld -c exporter /bin/bash 
curl localhost:9187/metrics

# replica pod -> worked 
kubectl exec -ti infra-qtxf-5686b758dc-5xkzq -c exporter /bin/bash 
curl localhost:9187/metrics
```

**Comment by jkatz** [comment]
## Primary
```
$ kubectl -n jkatz exec -it infra-6cc5bfcbfd-2lb5p -c exporter  -- curl localhost:9187/metrics
# HELP ccp_archive_command_status_seconds_since_last_fail Seconds since the last recorded failure of the archive_command
# TYPE ccp_archive_command_status_seconds_since_last_fail gauge
ccp_archive_command_status_seconds_since_last_fail{server="127.0.0.1:5432"} 0
# HELP ccp_backrest_last_diff_backup_time_since_completion_seconds Seconds since the last completed full or differential backup. Differential is always based off last full.
# TYPE ccp_backrest_last_diff_backup_time_since_completion_seconds gauge
ccp_backrest_last_diff_backup_time_since_completion_seconds{config_file="",server="127.0.0.1:5432",stanza="db"} 2963.451177
# ... truncated
```

## Replica
```
[vagrant@master postgres-operator]$ kubectl -n jkatz exec -it infra-bdxr-574bb8d6-9l8s6 -c exporter  -- curl localhost:9187/metrics
# HELP ccp_archive_command_status_seconds_since_last_fail Seconds since the last recorded failure of the archive_command
# TYPE ccp_archive_command_status_seconds_since_last_fail gauge
ccp_archive_command_status_seconds_since_last_fail{server="127.0.0.1:5432"} 0
# HELP ccp_backrest_last_diff_backup_time_since_completion_seconds Seconds since the last completed full or differential backup. Differential is always based off last full.
# TYPE ccp_backrest_last_diff_backup_time_since_completion_seconds gauge
ccp_backrest_last_diff_backup_time_since_completion_seconds{config_file="",server="127.0.0.1:5432",stanza="db"} 2613.347504
# HELP ccp_backrest_last_full_backup_time_since_completion_seconds Seconds since the last completed full backup
# TYPE ccp_backrest_last_full_backup_time_since_completion_seconds gauge
ccp_backrest_last_full_backup_time_since_completion_seconds{config_file="",server="127.0.0.1:5432",stanza="db"} 2613.348103
# Truncated...
```

**Comment by lphan-clv** [comment]
Hmm there it's still the same for me after upgrade to 4.6.1:

```
# /metrics timed out
bash-4.4$ curl -vvv localhost:9187/metrics
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9187 (#0)
> GET /metrics HTTP/1.1
> Host: localhost:9187
> User-Agent: curl/7.61.1
> Accept: */*
>

^C

bash-4.4$ curl -vvv localhost:9187
* Rebuilt URL to: localhost:9187/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 9187 (#0)
> GET / HTTP/1.1
> Host: localhost:9187
> User-Agent: curl/7.61.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/html; charset=UTF-8
< Date: Fri, 05 Mar 2021 15:23:02 GMT
< Content-Length: 148
<
<html>
	<head><title>Postgres exporter</title></head>
	<body>
	<h1>Postgres exporter</h1>
	<p><a href='/metrics'>Metrics</a></p>
	</body>
	</html>
* Connection #0 to host localhost left intact
	bash-4.4$

```

**Comment by lphan-clv** [comment]
@jkatz : It might be a bug here, when I removed s3 backup settings, it worked fine:

Deploy command :

```

pgo create cluster infra \
    --pgbouncer \
    --service-type=LoadBalancer \
    --server-ca-secret=postgresql-ca \
    --server-tls-secret=infra-tls-keypair \
    --pgbouncer-tls-secret=infra-tls-keypair \
    --replication-tls-secret=infra-rep-tls-keypair \
    --pgbouncer-replicas=2 \
    --replica-count=1 \
    --pvc-size=10Gi \
    --metrics

```

Metrics ouptut:

```

bash-4.4$ curl -s localhost:9187/metrics | head -10
# HELP ccp_archive_command_status_seconds_since_last_fail Seconds since the last recorded failure of the archive_command
# TYPE ccp_archive_command_status_seconds_since_last_fail gauge
ccp_archive_command_status_seconds_since_last_fail{server="127.0.0.1:5432"} 0
# HELP ccp_connection_stats_active Total non-idle connections
# TYPE ccp_connection_stats_active gauge
ccp_connection_stats_active{server="127.0.0.1:5432"} 7
# HELP ccp_connection_stats_idle Total idle connections
# TYPE ccp_connection_stats_idle gauge
ccp_connection_stats_idle{server="127.0.0.1:5432"} 1
# HELP ccp_connection_stats_idle_in_txn Total idle in transaction connections

```

**Comment by jkatz** [comment]
I am not reproducing this. Using a similar command:

```
pgo create cluster infra \
    --pgbouncer \
    --service-type=LoadBalancer \
    --server-ca-secret=postgresql-ca \
    --server-tls-secret=hippo.tls \
    --pgbouncer-tls-secret=hippo.tls \
    --replication-tls-secret=hippo.tls \
    --pgbouncer-replicas=2 \
    --replica-count=1 \
    --pvc-size=10Gi \
    --metrics \
    --pgbackrest-storage-type s3 \
    --pgbackrest-s3-bucket=REDACTED \
    --pgbackrest-s3-endpoint=REDACTED \
    --pgbackrest-s3-region=REDACTED \
    --pgbackrest-s3-key=REDACTED \
    --pgbackrest-s3-key-secret=REDACTED
```

### Primary

`kubectl exec -it infra-874bfb975-knxtt -c exporter -- curl http://localhost:9187/metrics`

```
# HELP ccp_archive_command_status_seconds_since_last_fail Seconds since the last recorded failure of the archive_command
# TYPE ccp_archive_command_status_seconds_since_last_fail gauge
ccp_archive_command_status_seconds_since_last_fail{server="127.0.0.1:5432"} 0
# HELP ccp_connection_stats_active Total non-idle connections
# TYPE ccp_connection_stats_active gauge
ccp_connection_stats_active{server="127.0.0.1:5432"} 6
# HELP ccp_connection_stats_idle Total idle connections
# TYPE ccp_connection_stats_idle gauge
ccp_connection_stats_idle{server="127.0.0.1:5432"} 2
# HELP ccp_connection_stats_idle_in_txn Total idle in transaction connections
```

### Replica

`kubectl exec -it infra-znmz-775778698f-6npnm -c exporter -- curl http://localhost:9187/metrics`

```
# HELP ccp_archive_command_status_seconds_since_last_fail Seconds since the last recorded failure of the archive_command
# TYPE ccp_archive_command_status_seconds_since_last_fail gauge
ccp_archive_command_status_seconds_since_last_fail{server="127.0.0.1:5432"} 0
# HELP ccp_connection_stats_active Total non-idle connections
# TYPE ccp_connection_stats_active gauge
ccp_connection_stats_active{server="127.0.0.1:5432"} 6
# HELP ccp_connection_stats_idle Total idle connections
# TYPE ccp_connection_stats_idle gauge
ccp_connection_stats_idle{server="127.0.0.1:5432"} 2
# HELP ccp_connection_stats_idle_in_txn Total idle in transaction connections
```

It sounds like if you are having issues with connecting to S3, this is a configuration issue in your environment.

I don't believe there is a bug. If you are looking for additional support, please review the [Support](https://access.crunchydata.com/documentation/postgres-operator/latest/support/) page.

**Comment by lphan-clv** [comment]
@jkatz : My s3 worked fine. The first reason to let me I came to CrunchyData postgres-operator is for letting me easier when using pgbackrest.  I tested it first when my cluster booted up at the first time and see it worked well with my S3. 

Anyway, I will re up the old cluster with S3 supports, and show you its worked, let's wait for me.

**Comment by jkatz** [comment]
The Operator does not automatically delete data from S3. If you are creating a cluster, deleting a cluster, and recreating a cluster and pushing to the same bucket, then it will not work. I believe this is noted in the documentation.

Again, I don't believe there is a bug. If you are looking for additional support, please review the [Support](https://access.crunchydata.com/documentation/postgres-operator/latest/support/) page.

**Comment by lphan-clv** [comment]
@jkatz Yes I always remove all data from my S3 before booting a new cluster . One more thing is my S3 buit from Minio

About the replies, I just want to see there is a bug or not. Let's see. And thanks for your suggestion about Support page!

**Comment by lphan-clv** [comment]
@jkatz Please see my reproduced. If you need any more informations, I can support.

**Commands**

```

> pgo create cluster infra \
    --pgbouncer \
    --service-type=LoadBalancer \
    --server-ca-secret=postgresql-ca \
    --server-tls-secret=infra-tls-keypair \
    --pgbouncer-tls-secret=infra-tls-keypair \
    --replication-tls-secret=infra-rep-tls-keypair \
    --pgbouncer-replicas=2 \
    --replica-count=1 \
    --pvc-size=10Gi \
    --pgbackrest-storage-type=s3 \
    --pgbackrest-s3-bucket=psql-infra-backup \
    --pgbackrest-s3-endpoint=XXX \
    --pgbackrest-s3-key=YYY \
    --pgbackrest-s3-key-secret=ZZZ \
    --pgbackrest-s3-region=eu-cenral-1 \
    --pgbackrest-s3-uri-style=path \
    --pgbackrest-s3-verify-tls=false \
    --metrics


created cluster: infra
workflow id: 2f6f1186-3345-49c8-b18b-be99a4530f59
database name: infra
users:
	username: testuser password: aaa 


> pgo backup infra
created Pgtask backrest-backup-infra

> pgo show backup  infra

cluster: infra
storage type: s3

stanza: db
    status: ok
    cipher: none

    db (current)
        wal archive min/max (13-1)

        full backup: 20210305-160905F
            timestamp start/stop: 2021-03-05 23:09:05 +0700 +07 / 2021-03-05 23:09:50 +0700 +07
            wal start/stop: 000000010000000000000002 / 000000010000000000000002
            database size: 31.1MiB, backup size: 31.1MiB
            repository size: 3.8MiB, repository backup size: 3.8MiB
            backup reference list:

        incr backup: 20210305-160905F_20210305-161322I
            timestamp start/stop: 2021-03-05 23:13:22 +0700 +07 / 2021-03-05 23:13:30 +0700 +07
            wal start/stop: 000000010000000000000006 / 000000010000000000000006
            database size: 31.2MiB, backup size: 1.9MiB
            repository size: 3.8MiB, repository backup size: 262.4KiB
            backup reference list: 20210305-160905F

> 

```

**S3 content (I removed all things from this bucket before booting the cluster)**

Please don't case about 36.05GB, it's disk space for my 3 buckets 

![image](https://user-images.githubusercontent.com/67224217/110142657-f9c1d500-7e08-11eb-9e78-d94b636c7bdd.png)

**Comment by dakine1111** [comment]
Hi,
with 4.6.1 and 4.6.2 I've the same problem.
OCP4.6
Create a cluster with S3 as pgbackrest-storage-type:
```
pgo create cluster postgresql01 \
  --replica-count 1 \
  -n shared-dev \
  --storage-config storageos50G \
  --replica-storage-config storageos50G \
  --pgbackrest-storage-type s3 \
  --cpu-limit 700m \
  --cpu 300m \
  --memory-limit 768Mi \
  --memory 256Mi
```
Cluster status:
```
cluster : postgresql01 (crunchy-postgres-ha:centos8-12.6-4.6.2)
        pod : postgresql01-77854f4bd9-ffp29 (Running) on ai03oswd2 (2/2) (primary)
                pvc: postgresql01 (47684Mi)
        pod : postgresql01-snco-77647475f-9cznc (Running) on ai03oswd0 (2/2) (replica)
                pvc: postgresql01-snco (47684Mi)
        resources : CPU: 300m Memory: 256Mi
        limits : CPU: 700m Memory: 768Mi
        deployment : postgresql01
        deployment : postgresql01-backrest-shared-repo
        deployment : postgresql01-snco
        service : postgresql01 - ClusterIP (172.30.132.106) - Ports (9187/TCP, 2022/TCP, 5432/TCP)
        service : postgresql01-replica - ClusterIP (172.30.228.68) - Ports (9187/TCP, 2022/TCP, 5432/TCP)
        pgreplica : postgresql01-snco
        labels : crunchy-pgha-scope=postgresql01 crunchy-postgres-exporter=true deployment-name=postgresql01 name=postgresql01 pg-cluster=postgresql01 pgo-version=4.6.2 pgouser=admin workflowid=1c1c9e67-9241-4d25-a244-2c34abb86f8f
```
Initial Backup:
```
cluster: postgresql01
storage type: s3

stanza: db
    status: ok
    cipher: none

    db (current)
        wal archive min/max (12-1)

        full backup: 20210426-125232F
            timestamp start/stop: 2021-04-26 14:52:32 +0200 CEST / 2021-04-26 14:53:00 +0200 CEST
            wal start/stop: 000000010000000000000002 / 000000010000000000000002
            database size: 31.4MiB, backup size: 31.4MiB
            repository size: 3.8MiB, repository backup size: 3.8MiB
            backup reference list: 
```
Leader:
```
oc -n shared-dev rsh -c exporter postgresql01-77854f4bd9-ffp29 curl -v localhost:9187/metrics
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9187 (#0)
> GET /metrics HTTP/1.1
> Host: localhost:9187
> User-Agent: curl/7.61.1
> Accept: */*
> 
^Ccommand terminated with exit code 130
```
```
oc -n shared-dev logs -c exporter postgresql01-77854f4bd9-ffp29
...
time="2021-04-26T13:11:59Z" level=error msg="queryNamespaceMappings returned 4 errors" source="postgres_exporter.go:1474"
2021/04/26 13:11:59 http: multiple response.WriteHeader calls
time="2021-04-26T13:35:53Z" level=info msg="Error running query on database \"127.0.0.1:5432\": ccp_backrest_last_full_backup pq: invalid input syntax for type json" source="postgres_exporter.go:1356"
time="2021-04-26T13:36:53Z" level=info msg="Error running query on database \"127.0.0.1:5432\": ccp_backrest_last_info pq: invalid input syntax for type json" source="postgres_exporter.go:1356"
time="2021-04-26T13:37:54Z" level=info msg="Error running query on database \"127.0.0.1:5432\": ccp_backrest_last_full_backup pq: invalid input syntax for type json" source="postgres_exporter.go:1356"
```
```
oc -n shared-dev rsh postgresql01-77854f4bd9-ffp29 pgbackrest info
command terminated with exit code 95
```
```
psql -h 127.0.0.1 -U postgres -c "select * from monitor.pgbackrest_info"
 config_file | data | gather_timestamp 
-------------+------+------------------
(0 rows)
```
Replica:
```oc -n shared-dev rsh -c exporter postgresql01-snco-77647475f-9cznc curl -v localhost:9187/metrics
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9187 (#0)
> GET /metrics HTTP/1.1
> Host: localhost:9187
> User-Agent: curl/7.61.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Type: text/plain; version=0.0.4; charset=utf-8
< Date: Mon, 26 Apr 2021 13:33:48 GMT
< Transfer-Encoding: chunked
< 
# HELP ccp_archive_command_status_seconds_since_last_fail Seconds since the last recorded failure of the archive_command
# TYPE ccp_archive_command_status_seconds_since_last_fail gauge
ccp_archive_command_status_seconds_since_last_fail{server="127.0.0.1:5432"} 0
# HELP ccp_connection_stats_active Total non-idle connections
# TYPE ccp_connection_stats_active gauge
ccp_connection_stats_active{server="127.0.0.1:5432"} 5
...
```
```
oc -n shared-dev rsh postgresql01-snco-77647475f-9cznc pgbackrest info
command terminated with exit code 95
```
As a conclusion metrics seems to be broken together with S3 pgbackrest. I can reproduce this behavior on a second OCP.

**Comment by jkatz** [comment]
https://github.com/CrunchyData/postgres-operator/issues/2349#issuecomment-814201730

**Comment by dakine1111** [comment]
But unfortunately on a 8d old (upgraded) cluster metrics are still not working correctly :-(
`postgresql05-697bfd494b-fvrcc                             2/2     Running     0          8d`
```
oc -n shared-prod rsh -c exporter postgresql05-697bfd494b-fvrcc curl -v localhost:9187/metrics
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9187 (#0)
> GET /metrics HTTP/1.1
> Host: localhost:9187
> User-Agent: curl/7.61.1
> Accept: */*
> 
^Ccommand terminated with exit code 130
```

**Comment by jkatz** [comment]
Do you have any output from the Postgres logs in `/pgdata/<clustername>/pg_log`?

**Comment by lphan-clv** [comment]
Hmm, I hope we can resolve this issue, I must stop using postgres-operator because of the current issue.

**Comment by dakine1111** [comment]
Good hint.
```
2021-04-27 06:19:16.555 UTC [245] ERROR:  invalid input syntax for type json
2021-04-27 06:19:16.555 UTC [245] DETAIL:  The input string ended unexpectedly.
2021-04-27 06:19:16.555 UTC [245] CONTEXT:  JSON data, line 1:
        COPY pgbackrest_info, line 1, column data: ""
        SQL statement "COPY monitor.pgbackrest_info (config_file, data) FROM program '/opt/crunchy/bin/postgres-ha/pgbackrest/pgbackrest_info.sh' WITH (format text,DELIMITER '|')"
        PL/pgSQL function monitor.pgbackrest_info(integer) line 21 at SQL statement
2021-04-27 06:19:16.555 UTC [245] STATEMENT:  WITH all_backups AS ( SELECT config_file , jsonb_array_elements(data) AS stanza_data FROM monitor.pgbackrest_info(10) ) , per_stanza AS ( SELECT config_file , stanza_data->>'name' AS stanza , jsonb_array_elements(stanza_data->'backup') AS backup_data FROM all_backups ) SELECT config_file , stanza , extract(epoch from (CURRENT_TIMESTAMP - max(to_timestamp((backup_data->'timestamp'->>'stop')::bigint)))) AS time_since_completion_seconds FROM per_stanza WHERE backup_data->>'type' IN ('full') GROUP BY config_file, stanza
ERROR: [095]: raised from remote-0 protocol on 'postgresql05-backrest-shared-repo': unable to verify certificate presented by 'minio02-shared.apps.ocp1.mydomain.de:443': [19] self signed certificate in certificate chain
```
BackrestS3VerifyTLS is disabled:
```
BackrestStorage: storageos100G
BackupStorage: storageos100G
BasicAuth: ""
Cluster:
  BackrestPort: 2022
  BackrestS3Bucket: pgbackup
  BackrestS3Endpoint: minio02-shared.apps.ocp1.mydomain.de
  BackrestS3Region: minio
  BackrestS3URIStyle: path
  BackrestS3VerifyTLS: "false"
  Badger: false
```

**Comment by jkatz** [comment]
I need to verify this in the code, but it seems we may need to ensure that if one sets the "verify TLS" setting to false, that the script that is scraping the pgBackRest metrics honors it.

Now, I will suggest that the actual answer is to fix the TLS setup in your environment, but I do think that the metrics scraper should honor that setting.

**Comment by jkatz** [comment]
I was able to successfully verify this and proposing the fix in CrunchyData/crunchy-containers#1354.

I feel confident backpatching this down at least to 4.6, potentially to 4.5. I will close this issue once the patch is merged in.

Thanks for reporting!

**Comment by jkatz** [comment]
Patched cleanly down to 4.5.
